index.d.ts

```diff
@@ -98,6 +98,18 @@ declare class BasePrompt extends EventEmitter {
     render(): void;
 
     run(): Promise<any>;
+  
+    cursorHide: () => void;
+    state: {
+      size: number;
+      submitted: boolean;
+    };
```

Comment:
This could also be specified separately as:

```ts
  export class State {
    type: string
    name: string
    message: string
    header: string
    footer: string
    error: string
    hint: string
    input: string
    cursor: number
    index: number
    lines: number
    tick: number
    prompt: string
    buffer: string
    width: number
    symbols: any
    styles: any
    required: typeof Set
    cancelled: boolean
    submitted: boolean
    loading: boolean | 'choices'
    readonly status: 'pending' | 'cancelled' | 'submitted'
    _choices: types.Choice[]

    clone(): Omit<State, 'clone' | 'buffer'> & { buffer: Buffer }

    color: Function | any
  }
```